src/tools.ts

```diff
@@ -81,7 +81,11 @@ export class ToolsConfig {
                     } while (action === 'Download again');
 
                     if (action !== 'Cancel') {
-                        await Archive.unzip(toolDlLocation, path.resolve(Platform.getUserHomePath(), '.vs-tekton'), "");
+                        if (toolDlLocation.endsWith('.zip') || toolDlLocation.endsWith('.tar.gz')) {
```

Comment:
This if/else block is not required, because all the variants are handled inside Archive.unzip method. `filePrefix` will be ignored for non tar archives.
It should be just 
```
await Archive.unzip(toolDlLocation, toolCacheLocation, ToolsConfig.tool['tkn'].filePrefix);
```

Reply:
@sudhirverma  this is actually fine, because .gz should be really processed differently.